src/js/store/perfdata/actions.js

```diff
@@ -38,39 +47,52 @@ export const perfRESTFetch = (url: string, params?: {}): ThunkAction => (
   if (params) {
     url = `${url}&${queryString.stringify(params)}`;
   }
-  dispatch({ type: 'PERF_DATA_LOADING', payload: { url } });
-  apiFetch(url, {
-    method: 'GET',
-  }).then(payload => {
-    if (payload) {
-      if (!payload.error) {
-        return dispatch({ type: 'PERF_DATA_AVAILABLE', payload });
-      }
-      return dispatch({ type: 'PERF_DATA_ERROR', payload });
-    }
-    return undefined;
+  return perfREST_API({
+    dispatch,
+    _getState,
+    apiFetch,
+    prefix: 'PERF',
+    url,
+    checkValid: assertPerfData,
   });
 };
 
 export const perfREST_UI_Fetch = (): ThunkAction => (
   dispatch,
   _getState,
   { apiFetch },
-) => {
-  const url = '/api/testmethod_perf_UI';
-  dispatch({ type: 'UI_DATA_LOADING', payload: url });
-  apiFetch(url, {
-    method: 'GET',
-  }).then(payload => {
-    if (payload) {
-      if (!payload.error) {
-        const typedPayload: UIData = assertUIData(payload);
-        return dispatch({ type: 'UI_DATA_AVAILABLE', payload: typedPayload });
-      }
-      return dispatch({ type: 'UI_DATA_ERROR', payload });
+) =>
+  perfREST_API({
+    dispatch,
+    _getState,
+    apiFetch,
+    prefix: 'UI',
+    url: testmethod_perf_UI_url,
+    checkValid: assertUIData,
+  });
+
+// $FlowFixMe
+export const perfREST_API = ({
+  dispatch,
+  _getState,
+  apiFetch,
+  prefix,
+  url,
+  checkValid,
+}) => {
+  dispatch({ type: `${prefix}_DATA_LOADING`, payload: { url } });
+  return apiFetch(url, { method: 'GET' }).then(payload => {
+    try {
+      if (!payload){ throw new Error('No payload');} // prettier-ignore
+      if (payload.error){ throw new Error(`Server error: ${payload.error}`);} // prettier-ignore
```

Comment:
I think it'd be better to follow the format prettier wants here. I find single-line conditionals less readable even without the extra comment.